include/swift/TBDGen/TBDGen.h

```diff
@@ -88,10 +89,7 @@ struct TBDGenOptions {
   }
 };
 
-void enumeratePublicSymbols(FileUnit *module, llvm::StringSet<> &symbols,
-                            const TBDGenOptions &opts);
-void enumeratePublicSymbols(ModuleDecl *module, llvm::StringSet<> &symbols,
-                            const TBDGenOptions &opts);
+llvm::StringSet<> getPublicSymbols(TBDGenDescriptor desc);
```

Comment:
`TBDGenDescriptor &&` please

Reply:
Why?

Reply:
This is a consuming use of the descriptor.

Reply:
Wouldn't it be valid to re-use a descriptor? e.g make a call to get the public symbols, make a call to emit the TBD file

Reply:
Do we have usages like that? I mean, you raise a good point so this is definitely just a minor nit you can ignore if you want.

Reply:
Not that I'm aware of, though it seems like a reasonable thing to want to do, so I think I'll leave this as-is for now.